### PR TITLE
Add sltb009a BSP

### PR DESCRIPTION
**Pull request by perigoso**
Board support for the EFM32GG12 Dev Board

For use with the efm32 family driver added in #703

### CODE REVIEW

File: hw/bsp/sltb009a/board.mk [hunk 1]
```diff
@@ -0,0 +1,45 @@
+CFLAGS += \
```
Comment (hathach) [inline hw/bsp/sltb009a/board.mk:1]: please add the following line, that will help to fix the CI. I have tried may the build to fetch needed submodules (if not already) without user have to do it manually or init it all.

```
DEPS_SUBMODULES += hw/mcu/silabs/cmsis-dfp-efm32gg12b
```

Comment (perigoso) [reply hw/bsp/sltb009a/board.mk:1]: I did add that, it's a bit lower down.

Comment (hathach) [reply hw/bsp/sltb009a/board.mk:1]: ah right, it is probably need to have the #703 merged first (and/or) update/rebase aftewrads.

**Comment by hathach** [comment]
#703 is merged, could you rebase/erge from master to see it could fix the ci.

**Comment by perigoso** [comment]
ah right, I almost forgot about this issue

The silabs cmsis has both a assembly startup and a C startup, on the same dir, for some reason this generates some conflict (im only including the assembly startup), the only way i got this to work was by deleting the c startup

**Comment by perigoso** [comment]
I don't really understand this issue

```
[perigoso@perigoso-desktop hid_generic_inout]$ make BOARD=sltb009a print-SRC_C
SRC_C = hw/mcu/silabs/cmsis-dfp-efm32gg12b/Device/SiliconLabs/EFM32GG12B/Source/system_efm32gg12b.c src/portable/silabs/efm32/dcd_efm32.c hw/bsp/board.c hw/bsp/sltb009a/sltb009a.c examples/device/hid_generic_inout/src/main.c examples/device/hid_generic_inout/src/usb_descriptors.c src/tusb.c src/common/tusb_fifo.c src/device/usbd.c src/device/usbd_control.c src/class/audio/audio_device.c src/class/cdc/cdc_device.c src/class/dfu/dfu_rt_device.c src/class/hid/hid_device.c src/class/midi/midi_device.c src/class/msc/msc_device.c src/class/net/net_device.c src/class/usbtmc/usbtmc_device.c src/class/vendor/vendor_device.c
[perigoso@perigoso-desktop hid_generic_inout]$ make BOARD=sltb009a print-SRC_S
SRC_S = hw/mcu/silabs/cmsis-dfp-efm32gg12b/Device/SiliconLabs/EFM32GG12B/Source/GCC/startup_efm32gg12b.s
```

the file it complains about isnt even supposed to be compiled
```
/home/perigoso/Development/tinyusb/hw/mcu/silabs/cmsis-dfp-efm32gg12b/Device/SiliconLabs/EFM32GG12B/Source/GCC/startup_efm32gg12b.c:44:17: error: conflicting type qualifiers for '__copy_table_start__'
   44 | extern uint32_t __copy_table_start__;
      |                 ^~~~~~~~~~~~~~~~~~~~
```

**Comment by perigoso** [comment]
I think this might be some oversight on the buildsystem
to clarify
the issue here is that there are two files named the same in the silabs cmsis, one in C and the other in assembly
`hw/mcu/silabs/cmsis-dfp-efm32gg12b/Device/SiliconLabs/EFM32GG12B/Source/GCC/startup_efm32gg12b.S`
`hw/mcu/silabs/cmsis-dfp-efm32gg12b/Device/SiliconLabs/EFM32GG12B/Source/GCC/startup_efm32gg12b.c`
the only file being included is the assembly one, and thats the only one we want to compile, but it seems the project tries to compile the c file too

**Comment by perigoso** [comment]
I've identified the issue I believe, in rules. Mk.

Can I make some minor changes in the buildsystem?

**Comment by perigoso** [comment]
Needs #755 fixed for successful build

**Comment by perigoso** [comment]
my dev board arrived today, will test this tomorrow

**Comment by hathach** [comment]
you probably miss the freertos header include for the efm https://github.com/hathach/tinyusb/blob/master/hw/bsp/board_mcu.h . It is not too obvious and well-placed currently. I plan to refactor it later on

**Comment by perigoso** [comment]
I just tested the dev board and the board support code i wrote works :)
I have not tested the buttons or implemented uart, but the devices enumerates and theres visual feedback from the leds

**Comment by perigoso** [comment]
If everything looks good to you you can merge, feel free to request changes

**Review by hathach — APPROVED** [review]
Superb !! Thank you very much for your PR. It looks perfect now

**Pull request merged**